### PR TITLE
feat: add types

### LINKED ISSUE #34: RFE: Please add support for TypeScript
It would be great to have an option to use `adapter-github-actions` in Typescript.

Could you please add `.d.ts` file. Thank you.

### PR DESCRIPTION
**Pull request by jamacku**
This should resolve #34

Please note, that this PR doesn't cover `ProbotExports`: https://github.com/probot/adapter-github-actions/blob/8e1f0f86d200547d554fbdbdf01088ec13236fc0/index.js#L1-L6

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { ApplicationFunction } from "probot/lib/types";
```
Comment (gr2m) [inline index.d.ts:1]: I think this should work

```suggestion
import { ApplicationFunction } from "probot";
```

Can you check it please?

Comment (jamacku) [reply index.d.ts:1]: I tried that at first, but my IDE was saying: `Module '"probot"' has no exported member 'ApplicationFunction'.ts(2305)` And When I checked probot's `index.d.ts` there is only export for `ApplicationFunctionOptions` but no `ApplicationFunction` type.

Comment (gr2m) [reply index.d.ts:1]: I think we need to update this line first in probot:
https://github.com/probot/probot/blob/bb068d9ae2407df8a07ffa64dc7d7fb8f01f097a/src/index.ts#L24

Comment (jamacku) [reply index.d.ts:1]: Sure, I'll create PR.

Comment (gr2m) [reply index.d.ts]: Already merged, thanks! Now you'll have to bump the minimum version of probot to `^12.1.4`

https://github.com/probot/adapter-github-actions/blob/8e1f0f86d200547d554fbdbdf01088ec13236fc0/package.json#L34

Note there were some breaking changes in v12, I'm not sure if any of them affect this adapter:
https://github.com/probot/probot/releases/tag/v12.0.0

Comment (jamacku) [reply index.d.ts]: Updated, it seems like these breaking changes doesn't affect adapter, but I found out that there is failing one unit test.

It fails here:

https://github.com/probot/adapter-github-actions/blob/8e1f0f86d200547d554fbdbdf01088ec13236fc0/test/index.test.js#L99

With message:
```js
expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

       97 |     process.stdout.write = origWrite;
       98 |
    >  99 |     expect(output[2].startsWith('::error::{"error":"nope"}%0A{%0A')).toBe(true);
          |                                                                      ^
      100 |
      101 |     expect(mock.activeMocks()).toStrictEqual([]);
      102 |   });

      at Object.<anonymous> (test/index.test.js:99:70)
```

Is it possible that error message format had changed or am I missing something?

**Comment by gr2m** [comment]
Děkuju Jan!

**Comment by jamacku** [comment]
Thank you for your work and this awesome tool! :+1:

**Pull request merged**